code/game/objects/items/weapons/cigs_lighters.dm

```diff
@@ -490,33 +490,77 @@ CIGARETTE PACKETS ARE IN FANCY.DM
 	description_fluff = "The legalization of certain substances propelled the sale of rolling papers through the roof. Now almost every Trans-stellar produces a variety, often of questionable quality."
 	icon = 'icons/obj/cigarettes.dmi'
 	icon_state = "cig paper"
+	volume = 25
+
+/obj/item/weapon/reagent_containers/rollingpaper/blunt
+	name = "blunt wrap"
+	desc = "A small piece of easily flammable paper similar to that which encases cigars. It's made out of tobacco, bigger than a standard rolling paper, and will last longer."
+	icon_state = "blunt paper"
+	volume = 45
 
-/obj/item/weapon/rollingpaper/attackby(obj/item/weapon/W as obj, mob/user as mob)
+/obj/item/weapon/reagent_containers/rollingpaper/attackby(obj/item/weapon/W as obj, mob/user as mob)
 	if (istype(W, /obj/item/weapon/reagent_containers/food/snacks))
 		var/obj/item/weapon/reagent_containers/food/snacks/grown/G = W
-		if (!G.dry)
-			to_chat(user, "<span class='notice'>[G] must be dried before you roll it into [src].</span>")
+		if (!G.dry)                                                                                          //This prevents people from just stuffing cheeseburgers into their joint
+			to_chat(user, "<span class='notice'>[G.name] must be dried before you add it to [src].</span>")
 			return
-		var/obj/item/clothing/mask/smokable/cigarette/joint/J = new /obj/item/clothing/mask/smokable/cigarette/joint(user.loc)
-		to_chat(usr, "<span class='notice'>You roll the [G.name] into a joint!</span>")
-		J.add_fingerprint(user)
+		if (G.reagents.total_volume + src.reagents.total_volume > src.reagents.maximum_volume)               //Check that we don't have too much already in the paper before adding things
+			to_chat(user, "<span class='warning'>The [src] is too full to add [G.name].</span>")
+			return
+		if (src.reagents.total_volume == 0)
+			if (istype(src, /obj/item/weapon/reagent_containers/rollingpaper/blunt))                         //update the icon if this is the first thing we're adding to the paper
+				src.icon_state = "blunt_full"
+			else
+				src.icon_state = "paper_full"
+		to_chat(user, "<span class='notice'>You add the [G.name] to the [src.name].</span>")
+		src.add_fingerprint(user)
 		if(G.reagents)
-			G.reagents.trans_to_obj(J, G.reagents.total_volume)
-		J.name = "[G.name] joint"
-		J.desc = "A joint lovingly rolled and filled with [G.name]. Blaze it."
+			G.reagents.trans_to_obj(src, G.reagents.total_volume)                                            //adds the reagents from the plant into the paper
+		user.drop_from_inventory(G)
 		qdel(G)
+
+/obj/item/weapon/reagent_containers/rollingpaper/attack_self(mob/living/user)
+	if(!src.reagents)                                                                                        //don't roll an empty joint
+		to_chat(user, "<span class='warning'>There is nothing in [src]. Add something to it first.</span>")
+		return
+	if (istype(src, /obj/item/weapon/reagent_containers/rollingpaper/blunt))

```

Comment:
This would be better as an override on `/obj/item/weapon/reagent_containers/rollingpaper/blunt`. Generally speaking if you're ever doing `istype(src)` your design needs reworking.

Reply:
You could also set a var on `/obj/item/weapon/reagent_containers/rollingpaper` like `var/joint_type = /obj/item/clothing/mask/smokable/cigarette/joint` and set it to the blunt type on `/rollingpaper/blunt`.